blocklist-client/src/config.rs

```diff
@@ -32,6 +32,21 @@ impl Settings {
         let env = Environment::with_prefix("BLOCKLIST_CLIENT").separator("__");
         cfg.merge(env)?;
         let settings: Settings = cfg.try_into()?;
+
+        settings.validate()?;
+
         Ok(settings)
     }
+
+    fn validate(&self) -> Result<(), ConfigError> {
+        if self.server.host.is_empty() {
+            return Err(ConfigError::Message("Host cannot be empty".to_string()));
+        }
+        if !(1..=65535).contains(&self.server.port) {
```

Comment:
I think this can be simplified to `port != 0`.

Alternatively, maybe we could try to parse the `ServerConfig` field(s) as `std::net::SocketAddr`? I think this would require writing a custom `deserialize_with` function or something like that since they do not implement `Deserialize`.

Reply:
Actually, I'm not entirely sure we should check for port zero, even though the IANA lists the port as [reserved](http://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml). I think I'd skip this check entirely.